Classes/Promise.swift

```diff
@@ -89,32 +71,23 @@ extension Array {
 public class PromiseCancelToken {
     
     public typealias Canceled = () -> Void
-    
-    internal let cancelFlag: AtomicCancel = AtomicCancel()
+    var workItem: DispatchWorkItem? = DispatchWorkItem(block: {}) // we can't inherit so we will use Decorator Pattern instead
```

Comment:
I suggest to use enforceQoS: DispatchWorkItemFlags it should increase priority for cancelation tasks.